docs/front-end/web-components.md

```diff
@@ -0,0 +1,85 @@
+# Web Components
+
+The [WHATWG](https://html.spec.whatwg.org/multipage/custom-elements.html) and [WICG](https://github.com/WICG/webcomponents) maintain suite of web technologies and specifications including (but not limited to)
+
+- [Custom Elements](https://html.spec.whatwg.org/multipage/custom-elements.html#custom-elements)
+- [Shadow DOM](https://dom.spec.whatwg.org/#shadow-trees)
+- The [`<template>` element](https://html.spec.whatwg.org/multipage/scripting.html#the-template-element)
+- The [`<slot>` element](https://html.spec.whatwg.org/multipage/scripting.html#the-slot-element)
+- [`ElementInternals`](https://html.spec.whatwg.org/multipage/custom-elements.html#element-internals)
+- [CSS Custom Properties](https://drafts.csswg.org/css-variables/)
+- [Module scripts](https://html.spec.whatwg.org/#module-script)
+
+These are collectively referred to as "web components", and represent the browsers' native component model for web developers.
+
+Broadly speaking, every web component is a custom element - a specific tag-name which becomes associated with a class extending HTMLElement in JavaScript. Web components can also make use of the related technologies.
+
+## Recommended Components
+
+- [Lit](https://lit.dev) (library and framework for authoring components)
+- [Web Dev Server](https://modern-web.dev) / Web Test Runner (for unit tests and local dev)
+- [Custom Elements Manifest](https://github.com/webcomponents/custom-elements-manifest) (for ide support / docgen / codegen)
+- [PatternFly Elements](https://patternflyelements.org) Design System
+- [Carbon Web Components](https://web-components.carbondesignsystem.com/) Design System
+- [Web Components Community Group](https://github.com/w3c/webcomponents-cg/)
+
+## Guidance
+
+### Our Experiences Using Web Components at Red Hat
+
+The team at Red Hat has been involved in several Web Component projects and have learned a good amount about Web Components and how to write them
+
+Two Web Component "Systems" at Red Hat are
+
+- [PatternFly Elements](https://patternflyelements.org/) (aka PFE)  (“upstream”)
+- [Red Hat Design System](https://ux.redhat.com/) (aka RHDS) (“downstream”)
+
+### Advantages and Ideal Use Cases
```

Comment:
We should add a section here on architecture:

- state management
- SPA routing
- data fetching

With an emphasis on the concept that web components are agnostic to all those things, and thus can integrate with most or all. For example, there are implementations of redux, mobx, apollo, etc etc for web components